### ISSUE TITLE
Not able to identify locators in Appium for a pop up box

### ISSUE THREAD

**Issue by sudhanshuprakash** [microsoft/WinAppDriver#1262]
I am trying to automate a word application using Appium. I have to insert a line or rectangle using appium. When I click on shapes option I am getting all the required shapes but I am not able to inspect the element. Instead, when I click on the shape icon(line or rectangle), the element which is getting selected is one just behind it. Although using points option I can click on the element but I am not able to click on shape element using any locator. So is there any way I can get the location address. Do I have to switch frames, if so how? I have attached an image for a better understanding of the problem. In this image when I am clicking on an arrow icon I am not able to get the details instead of that the element that is getting selected in the one just behind it. Here is a link of the issue: "https://i.stack.imgur.com/JPKPL.png"

Environment
Appium version (or git revision) that exhibits the issue: 1.18.0
Last Appium version that did not exhibit the issue (if applicable):
Desktop OS/version used to run Appium: Windows 10
Node.js version (unless using Appium.app|exe): 12.8.1
Npm or Yarn package manager:
Mobile platform/version under test: 7.1.1
Real device or emulator/simulator: Real Device
Appium CLI or Appium.app|exe: Appium.exe
I am using a tab for my testing.
Asked this issue in the git hub group appium/appium but got this reply: https://github.com/appium/appium/issues/14646#issuecomment-672734606

**Comment by anunay1** [comment]
Is it msword?

**Comment by sudhanshuprakash** [comment]
No. It is not MS Word but a different mobile application for word. What I have shown here is the page which is shown when I click on Start Inspector Session in appium desktop. No, I don't think that it is part of the same tree structure as all these shape options come after I click on shapes on the menu bar. I am not able to switch to this frame of pop-up and neither identify the element locations.

**Comment by anunay1** [comment]
Then you need to switch to the opened window,I don't know which language binding you are using, but this is how you do in C#: 

`
DesiredCapabilities desktopCapabilities = new DesiredCapabilities();
                desktopCapabilities.SetCapability("app", "Root");
                desktopSession = new WindowsDriver<WindowsElement>(new Uri(WindowsApplicationDriverUrl), desktopCapabilities);
                Assert.IsNotNull(desktopSession);

                var RSWindow = desktopSession.FindElementByName("You pop up window title");
                var RSTopLevelWindowHandle = RSWindow.GetAttribute("NativeWindowHandle");
                RSTopLevelWindowHandle = (int.Parse(RSTopLevelWindowHandle)).ToString("x");

                DesiredCapabilities appcapabilities = new DesiredCapabilities();
                appcapabilities.SetCapability("appTopLevelWindow", RSTopLevelWindowHandle);
                driver = new WindowsDriver<WindowsElement>(new Uri(WindowsApplicationDriverUrl), appcapabilities);

`
You can get the window title from inspect.exe.

**Comment by anunay1** [comment]
> No. It is not MS Word but a different mobile application for word. What I have shown here is the page which is shown when I click on Start Inspector Session in appium desktop. No, I don't think that it is part of the same tree structure as all these shape options come after I click on shapes on the menu bar. I am not able to switch to this frame of pop-up and neither identify the element locations.

Its not necessary that if the shapes comes after you click is part of the same window, check in inspect, it will give you the tree structure.

**Comment by sudhanshuprakash** [comment]
@anunay1 I don't think I was clear with my question. Now, I am doing my automation in a tablet and I am automating a mobile application. I tried using inspect.exe to identify the elements but there was no description for that although, all the menu bar options of the Appium window were present. I think this image https://i.stack.imgur.com/a0232.png will make it a bit more clear.1) I click on the shapes button. 2) A box appears with all the shape icons. 3) I want to find the locator of shape icons so I click on that (for eg. a straight arrow). Now what is happening is the whole page gets selected(highlighted in blue in the image) and its details are available in the selected element window.
What I want is just the details of one shape icon which is present under the shapes button.

**Comment by anunay1** [comment]
Is the app under your control, ask the development team to expose the controls, or else you will not be able to use it.

**Comment by sudhanshuprakash** [comment]
Okay. What exactly should I ask them?

**Comment by anunay1** [comment]
Ask them to implement automation properties.
